bucket-registry.go

```diff
@@ -29,54 +29,117 @@ import (
 // `realpath` or `fcntl(F_GETPATH)`) and use the canonical path as the key.
 // Unfortunately Go doesn't seem to have an API for that.
 
-var bucketRegistry = map[string][]*Bucket{} // Maps URL to slice of Buckets at that URL
-var bucketRegistryMutex sync.Mutex          // Thread-safe access to bucketRegistry
+type bucketRegistry struct {
+	byURL       map[string][]*Bucket
+	inMemoryRef map[string]*Bucket
```

Comment:
As discussed the names are a bit confusing if 'byURL' is actually the ref count, and 'inMemoryRef' is the lookup of bucket by URL.

Reply:
I picked names of `bucketCount` for refcount and `buckets` for the actual cached copy of the bucket.